src/Core/MySQL/PacketsProtocolBinary.cpp

```diff
@@ -0,0 +1,363 @@
+#include <Columns/IColumn.h>
+#include <Core/MySQL/IMySQLReadPacket.h>
+#include <Core/MySQL/IMySQLWritePacket.h>
+#include <Core/MySQL/PacketsProtocolBinary.h>
+#include "Common/LocalDate.h"
+#include "Common/LocalDateTime.h"
+#include "Columns/ColumnLowCardinality.h"
+#include "Columns/ColumnVector.h"
+#include "Columns/ColumnsDateTime.h"
+#include "Core/DecimalFunctions.h"
+#include "DataTypes/DataTypeDateTime64.h"
+#include "DataTypes/DataTypeLowCardinality.h"
+#include "DataTypes/DataTypeNullable.h"
+#include "DataTypes/DataTypesNumber.h"
+#include "Formats/FormatSettings.h"
+#include "IO/WriteBufferFromString.h"
+#include "base/DayNum.h"
+#include "base/Decimal.h"
+#include "base/types.h"
+
+namespace DB
+{
+
+namespace MySQLProtocol
+{
+
+    namespace ProtocolBinary
+    {
+        ResultSetRow::ResultSetRow(
+            const Serializations & serializations_, const DataTypes & data_types_, const Columns & columns_, int row_num_)
```

Comment:
In C++, you generally want to keep
- the order of (initialized) member variables in the class declaration (header file),
- the order of constructor parameters (l. 30) and
- the order in the initialization list (l. 31)

the same. Deviating from this rule opens the door to subtle bugs.